adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,44 @@
 #include "impl.h"
 #include <vector>
+#include <algorithm>
+#include <stack>
 
-const std::vector<std::pair<char, char>> brackets {
-    {'(', ')'},
-    {'[', ']'},
-    {'{', '}'}
-};
+namespace {
+    const std::vector<std::pair<char, char>> brackets {
+        {'(', ')'},
+        {'[', ']'},
+        {'{', '}'}
+    };
+}
+
+bool isValid(const std::string& source) {
+    std::stack<char> bracketsStack;
+
+    for(char symbol : source) {
+        auto itOpen = std::find_if(std::begin(brackets), std::end(brackets), [symbol](const std::pair<char, char> value) {
+                                                                             return symbol == value.first;
+                                                                         });
+        if(itOpen != std::end(brackets)) {
+            bracketsStack.push(symbol);
+            continue;
```

Comment:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es77-minimize-the-use-of-break-and-continue-in-loops